### PR TITLE
Fix opening url's with non-http schemes

### LINKED ISSUE #2395: Custom URL's in entry preview panel are launched incorrectly
## Expected Behavior
When URL field contains custom registered URL handler, for example rdp:// then system URL handler should be fired. It was working in 2.3.3 version.

## Current Behavior
In 2.3.4 version double click on url starting with custom url like rdp:// does nothing.

## Debug Info
KeePassXC - Wersja 2.3.4
Rewizja: 6fe821c

Biblioteki:
- Qt 5.11.1
- libgcrypt 1.8.3

System operacyjny: Windows 10 (10.0)
Architektura CPU: x86_64
Jądro: winnt 10.0.17134

Włączone rozszerzenia:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

### LINKED ISSUE #2427: Open terminal for ssh:// urls
ssh:// url - not working when double click on url field and when use shortcut or chose from contents menu
on 2.3.1 working but URL field on list was empty. 
## Expected Behavior

double click on url with ssh://{username}@ip:22 should open terminal automatically 
## Current Behavior

ssh:// url - not working when double click on url field and when use shortcut or chose from contents menu
## Steps to Reproduce (for bugs)

1. add to url eg. ssh://127.0.0.1:22 
2. Save 
3. try to Url open, double clic on URL or do CMD+U 

## Debug Info

KeePassXC - 2.3.4
Revision: master

Libraries:
- LIBS

Operating system: OSX 10.14 
CPU architecture: intel
Kernel: KERNEL

Enabled extensions:
- EXTENSIONS

### PR DESCRIPTION
**Pull request by SohamG**
Let the OS handle how to open the url.
Changed the openUrl() function to use a QUrl object, which has the appropriate scheme set.

[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

Fixes #2427
Fixes #2395 

## Testing strategy
Manually tested with ssh:// and https:// URLs on Linux 

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

Signed-off-by: sohamg <sohamg2@gmail.com>

### CODE REVIEW

File: src/gui/DatabaseWidget.cpp [hunk 1]
```diff
@@ -686,9 +686,12 @@ void DatabaseWidget::openUrlForEntry(Entry* entry)
             }
         }
     } else {
-        QString urlString = entry->webUrl();
+        QString urlString = entry->url();
         if (!urlString.isEmpty()) {
-            QDesktopServices::openUrl(urlString);
+            QString scheme = urlString.split(':')[0];
+            QUrl url = QUrl(urlString);
+            if (!scheme.isEmpty()) url.setScheme(scheme);
```
Comment (droidmonkey) [inline src/gui/DatabaseWidget.cpp]: wrap the code block with { }
Shouldn't QUrl already set the scheme based on a url of that structure?

**Comment by SohamG** [comment]
Fixed the if block ^^

I don't know if QUrl sets the scheme by default, I have no prior experience in Qt and the docs didn't mention that. I'll test if QUrl does this by default and update accordingly

**Comment by SohamG** [comment]
QUrl does indeed do this by default, thanks for the tip. Updated code accordingly.

**Comment by SohamG** [comment]
@droidmonkey Are any more changes required?

**Comment by droidmonkey** [comment]
Yes, I am working on this now

**Comment by SohamG** [comment]
Thanks, keep me posted

**Comment by droidmonkey** [comment]
OK this is ready to go, once CI completes I'll merge. I added the fix for the entry preview panel as well.

**Pull request merged**